### PR TITLE
Adding halt and scp

### LINKED ISSUE #290: Add some first class rambo commands
We should add `scp` and `halt` to the list of basic, first class rambo commands that pass through to the same vagrant commands.

### PR DESCRIPTION
**Pull request by nixjdm**
Pull after #291.

Fixes #290, #230 

Includes docs updates.

### CODE REVIEW

File: rambo/app.py [hunk 1]
```diff
@@ -323,7 +324,18 @@ def install_plugins(force=None, plugins=('all',)):
             if not force:
                 click.confirm('The plugin "%s" is not in our list of plugins. Attempt '
                           'to install anyway?' % plugin, abort=True)
-            _invoke_vagrant('plugin install %s' % plugin)
+            vagrant_general_command('plugin install %s' % plugin)
+
+def scp(ctx=None, locations=None):
+    assert(len(locations)==2)
```
Comment (smitty42) [inline rambo/app.py]: This is kinds of nasty. I'm assuming in the case the user would see an assertion that "len of locations not == 2". We know what the issue is, we could make that exception handling way nicer. Tell them how they probably screwed up and provide and example of what the input is supposed to look like.

File: rambo/app.py [hunk 2]
```diff
@@ -323,7 +324,18 @@ def install_plugins(force=None, plugins=('all',)):
...
+
+
+    if ':' in locations[0]: # [0] is remote, fix [1] which is local
```
Comment (smitty42) [inline rambo/app.py]: I want these variable names (location) to be assgined useful names like local and remote. I don't want to have to keep think about this stuff in terms of list indexes.

Comment (smitty42) [reply rambo/app.py]: Or  rather than remote / local, copy from, copy to. Which ever makes more since

File: rambo/cli.py [hunk 3]
```diff
@@ -119,32 +132,60 @@ def destroy_cmd(ctx):
...
-    all default vagrant plugins.
+    all default Vagrant plugins from host platform specific list.
     '''
     # If auth and plugins are both not specified, run both.
     if not plugins: # No args means all default plugins.
         plugins = ('all',)
     app.install_plugins(force, plugins)
 
 
+@cli.command('scp', context_settings=VAGRANT_CMD_CONTEXT_SETTINGS,
```
Comment (smitty42) [inline rambo/cli.py]: add examples

File: rambo/app.py (hunk 2 above)
Comment (nixjdm) [reply rambo/app.py]: :+1:

File: rambo/app.py (hunk 1 above)
Comment (nixjdm) [reply rambo/app.py]: Accidentally added. Will revise.

File: rambo/cli.py (hunk 3 above)
Comment (nixjdm) [reply rambo/cli.py]: Done

**Pull request merged**